### PR TITLE
Added declarations for rollup-plugin-css-only

### PR DESCRIPTION
**Pull request by SitamMatt**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/rollup-plugin-css-only/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Type definitions for rollup-plugin-css-only 3.1
+// Project: https://github.com/thgh/rollup-plugin-css-only (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (elibarzilay) [inline types/rollup-plugin-css-only/index.d.ts:2]: Leftover comment?

Comment (SitamMatt) [reply types/rollup-plugin-css-only/index.d.ts]: removed

File: types/rollup-plugin-css-only/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+/// <reference types="node" />
+import { Plugin, OutputBundle } from 'rollup';
+
+declare namespace css {
+    interface Options {
+        /**
+         *  All CSS files will be parsed by default, but you can also specifically include/exclude files
+         */
+        include?: ReadonlyArray<string | RegExp> | string | RegExp | null;
+        exclude?: ReadonlyArray<string | RegExp> | string | RegExp | null;
```
Comment (andrewbranch) [inline types/rollup-plugin-css-only/index.d.ts:15]: These are undocumented; sure you want to include them?

**Review by andrewbranch — APPROVED** [review]

File: types/rollup-plugin-css-only/index.d.ts (hunk 2 above)
Comment (SitamMatt) [reply types/rollup-plugin-css-only/index.d.ts:15]: I've copppied these include/exclude properties from another rollup plugin here, but after your comment I think it's better to fix this

File: types/rollup-plugin-css-only/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+         *  CSS files to exclude from being parsed
+         */
+        exclude?: ReadonlyArray<string | RegExp> | string | RegExp | null;
+        /**
+         * Callback that will be called ongenerate
+         */
+        output?:
+            | boolean
+            | string
+            | ((styles: string, styleNodes: Record<string, string>, bundle: OutputBundle) => void);
```
Comment (peterblazejewicz) [inline types/rollup-plugin-css-only/index.d.ts:25]:
```suggestion
            | ((styles: string, styleNodes: Record<string, string>, bundle: OutputBundle) => void)
            | null
            | undefined;
```
(readme):
```js
// Default behaviour is to write all styles to the bundle destination where .js is replaced by .css
  output: null
````
(undefined added to support exact optional properties from 4.4, but some js folks use null/undefined as unicorns)

File: types/rollup-plugin-css-only/rollup-plugin-css-only-tests.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import css = require('rollup-plugin-css-only');
+
+css({ output: 'bundle.css' });
+
+css({ output: false });
+
+css({ output: (styles, styleNodes, bundle) => {} });
+
+css();
```
Comment (peterblazejewicz) [inline types/rollup-plugin-css-only/rollup-plugin-css-only-tests.ts:9]:
```suggestion
css({
    output: null,
});

css();
```
(from readme)

File: types/rollup-plugin-css-only/index.d.ts (hunk 3 above)
Comment (SitamMatt) [reply types/rollup-plugin-css-only/index.d.ts:25]: ok, lgtm

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@SitamMatt thx!

**Comment by SitamMatt** [comment]
Ready to merge

**Pull request merged**